### DISCUSSION TITLE
Controller or its method is not found: \App\Controllers\LoginController::loginAction

### DISCUSSION THREAD

**Discussion by alfkoblischke** [codeigniter4/shield#1235] [Q&A]
I use shield for quite some time and it worked perfect. As I have a few Codeigniter 4 websites I had the crazy idea of running multiple Applications with one CodeIgniter Installation as descibed here: https://codeigniter.com/user_guide/general/managing_apps.html
My structure is more or less like in the description:
```
foo/
    app/
    public/
    tests/
    writable/
    env
    phpunit.xml.dist
    spark
bar/
    app/
    public/
    tests/
    writable/
    env
    phpunit.xml.dist
    spark
vendor/
    autoload.php
    codeigniter4/framework/
composer.json
composer.lock
```
On top of that I have a Localization in the form: https://example.com/en/home

All this works perfectly. But I cannot get Shield running. I get the Error Message: "Controller or its method is not found: \App\Controllers\LoginController::loginAction". I have copied the Auth, AuthGroups and AuthToken Files to the Config Folder of "foo" and "bar".
Can anybody give me a hint?

**Comment by grimpirate** [comment]
You have to modify your composer path to be one level higher as explained in the docs. The default value should be vendor/autoload.php but you want it to be ../vendor/autoload.php. This path should be updated within each config of your apps (app/Config/Constants.php). That's the likeliest culprit I can think of.

I set up all of my applications this way even if it's only one. If you can follow along with a Dockerfile you can see how I go about it [here](https://github.com/grimpirate/docker-ci-sqlite/blob/main/apache/Dockerfile).
